skimage/morphology/rolling_ball.py

```diff
@@ -0,0 +1,163 @@
+        `chunk_size`. Larger values will make processing faster, but will also
+        consume more memory.
+
+    Returns
+    -------
+    filtered_image : ndarray
+        The image with background removed.
+    background : ndarray
+        The background that was removed.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from skimage import data
+    >>> from skimage.morphology import rolling_ball
+
+    Subtract a black background from an example image
+
+    >>> image = data.coins()
+    >>> result, bg = rolling_ball(image, radius=100)
+
+    Subtract a white background from an example image
+
+    >>> image = data.page()
+    >>> result, bg = rolling_ball(image, radius=100, white_background=True)
+
+    References
+    ----------
+    .. [1] Sternberg, Stanley R. "Biomedical image processing." Computer 1
+           (1983): 22-34. :DOI:`10.1109/MC.1983.1654163`
+    """
+
+    try:
+        if len(radius) == 2:
+            spacial_radius = radius[0]
+            intensity_vertex = radius[1]
+    except TypeError:
+        spacial_radius = radius
+        intensity_vertex = radius
+    except IndexError:
+        # essentially radius need to have __len__ and __getitem__
+        # so that we can extract the values
+        raise ValueError(f"Radius must be a scalar or tuple-like.")
+
+    try:
+        spacial_radius = float(spacial_radius)
+    except ValueError:
+        raise ValueError(
+            f"Radius should be float or float tuple, "
+            f"was {type(spacial_radius)}"
+        )
+
+    try:
+        intensity_vertex = float(intensity_vertex)
+    except ValueError:
+        raise ValueError(
+            f"Radius should be float or float tuple, "
+            f"was {type(intensity_vertex)}")
+
+    if spacial_radius <= 0:
+        raise ValueError(
+            f"Spacial radius must be greater zero, was {spacial_radius}")
+
+    if intensity_vertex <= 0:
+        raise ValueError(
+            f"Intensity radius must be greater zero, was {intensity_vertex}")
+
+    try:
+        white_background = bool(white_background)
+    except ValueError:
+        raise ValueError(
+            f"white_background must be convertible to a boolean, "
+            f"was {type(white_background)}"
+        )
+
+    if not isinstance(image, np.ndarray):
+        raise ValueError(
+            f"input_img must be a np.ndarray, "
+            f"was {type(image)}"
+        )
+
+    if not issubclass(image.dtype.type, np.integer):
+        raise ValueError("Currently only integer images are supported.")
+
+    img = image.copy()
+    if white_background:
+        img = invert(img)
+
+    # assuming selem touches the umbra at (x,y) pre-compute the
+    # (relative) height of selem at the center
+    # (selem is a ball/spheroid)
+    spacial_upper_bound = int(np.ceil(spacial_radius))
+    L = np.arange(-spacial_upper_bound, spacial_upper_bound + 1)
+    X, Y = np.meshgrid(L, L)
+    distance = np.sqrt(X ** 2 + Y ** 2)
```

Comment:
Can you think of more meaningful names for `X`, `Y` and `L`? Using single-capital-letter variables feels a little unconventional to me.

Reply:
I made it lower capital xD and now use `x`, `y`, and `tmp`. X and Y are the coordinates of the ellipsoid in the x-y-plane; not sure what name would be more descriptive. Ideas?

Reply:
Thanks. Maybe `ellipsoid_x`, `ellipsoid_y`? It's not a huge issue but nice to have if something comes to mind. :wink: